src/details/ArborX_DetailsNode.hpp

```diff
@@ -22,19 +22,27 @@ namespace Details
 {
 struct Node
 {
-  KOKKOS_INLINE_FUNCTION
-  Node() = default;
+  KOKKOS_FUNCTION
+  constexpr Node() = default;
 
-  KOKKOS_INLINE_FUNCTION
-  Node(const Kokkos::pair<Node *, Node *> &c, const Box &bb)
-      : children(c)
-      , bounding_box(bb)
+  KOKKOS_FUNCTION constexpr bool isLeaf() const noexcept
```

Comment:
Please change `KOKKOS_FUNCTION` back to `KOKKOS_INLINE_FUNCTION` in all instances.

Reply:
Explicitly defaulted default constructor as well?

Reply:
Sure, if only for consistency. I think KOKKOS_INLINE_FUNCTION is more robust if at some point we move the body of the constructor/function out of the class. On the other hand, you can argue that you'll remember to switch `KOKKOS_FUNCTION` at that point.

Reply:
Done.